flink-container/src/main/java/org/apache/flink/container/entrypoint/ClassPathJobGraphRetriever.java

```diff
@@ -111,11 +117,10 @@ public JobGraph retrieveJobGraph(Configuration configuration) throws FlinkExcept
 	}
 
 	private PackagedProgram createPackagedProgram() throws FlinkException {
-		final String entryClass = getJobClassNameOrScanClassPath();
 		try {
-			final Class<?> mainClass = getClass().getClassLoader().loadClass(entryClass);
-			return new PackagedProgram(mainClass, programArguments);
-		} catch (ClassNotFoundException | ProgramInvocationException e) {
+			final String entryClass = getJobClassNameOrScanClassPath();
+			return new PackagedProgram(null, new ArrayList<>(getUserClassPaths()), entryClass, programArguments);
```

Comment:
Looks to me this PackagedProgram constructor behaves a bit differently to the previous one.
The previous one considers entryPointClassName "org.apache.flink.client.python.PythonDriver" or "org.apache.flink.client.python.PythonGatewayServer" to be python program, while the current used one only respects "org.apache.flink.client.python.PythonDriver". Not sure if it's a bug or expected. Would you take a check or confirm with related devs?

Reply:
I sync it with @sunjincheng121 . It is ok that we can check the two classes.